bindings/python/tests/test_sqlalchemy.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+import numpy as np
+from tests import *
+from sqlalchemy import create_engine, select, text, insert, delete
+from sqlalchemy import Integer, Index
+from pgvector_rs.sqlalchemy import Vector
+from sqlalchemy.orm import Session, DeclarativeBase, mapped_column, Mapped
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+class Document(Base):
+    __tablename__ = "tb_test_item"
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True)
+    embedding: Mapped[np.ndarray] = mapped_column(Vector(3))
+
+    def __repr__(self) -> str:
+        return f"{self.embedding}"
+
+
+@pytest.fixture(scope="module")
+def session():
+    """
+    Connect to the test db pointed by the URL. Can check more details
+    in `tests/__init__.py`
+    """
+
+    engine = create_engine(URL.replace("postgresql", "postgresql+psycopg"))
+
+    # ensure that we have installed pgvector.rs extension
+    with engine.connect() as conn:
+        conn.execute(text("CREATE EXTENSION IF NOT EXISTS vectors"))
+        conn.execute(text("DROP TABLE IF EXISTS tb_test_item"))
+        conn.commit()
+
+    with Session(engine) as session:
+        Document.metadata.create_all(engine)
+        try:
+            yield session
+        finally:
+            session.rollback()
+            Document.metadata.drop_all(engine)
+
+
+# =================================
+# Prefix functional tests
+# =================================
+
+
+@pytest.mark.parametrize("index_name,index_setting", TOML_SETTINGS.items())
+def test_create_index(session: Session, index_name: str, index_setting: str):
+    index = Index(
+        index_name,
+        Document.embedding,
+        postgresql_using="vectors",
+        postgresql_with={"options": index_setting},
+        postgresql_ops={"embedding": "l2_ops"},
+    )
+    index.create(session.bind)
+    session.rollback()
+
+
+@pytest.mark.parametrize("i,e", enumerate(INVALID_VECTORS))
+def test_invalid_insert(session: Session, i: int, e: np.ndarray):
+    try:
+        session.execute(insert(Document).values(id=i, embedding=e))
+    except:
+        session.rollback()
+    else:
+        session.rollback()
+        raise AssertionError(
+            "failed to raise invalid value error for {}th vector {}".format(i, e),
+        )
```

Comment:
You can rollback in `finally`. Besides, this `except` should be limited to several exceptions.